src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h

```diff
@@ -21,73 +21,38 @@
 
 #include <grpc/grpc_security.h>
 #include <grpc/support/port_platform.h>
+#include <vector>
+#include "src/core/tsi/s2a/s2a_constants.h"
 
 namespace experimental {
 
-enum s2a_supported_ciphersuite {
-  AES_128_GCM_SHA256 = 0,
-  AES_256_GCM_SHA384 = 1,
-  CHACHA20_POLY1305_SHA256 = 2,
-};
-
-/** An ordered list of ciphersuites for use by the S2A. **/
-typedef struct s2a_ciphersuite {
-  struct s2a_ciphersuite* next;
-  s2a_supported_ciphersuite cipher;
-} s2a_ciphersuite;
-
-/** The V-table for the grpc_s2a_credentials_options. **/
-typedef struct grpc_s2a_credentials_options_vtable {
-  grpc_s2a_credentials_options* (*copy)(
-      const grpc_s2a_credentials_options* options);
-  void (*destruct)(grpc_s2a_credentials_options* options);
-} grpc_s2a_credentials_options_vtable;
-
-/** The base struct for the S2A credentials options. The options contain an
- *  ordered list of S2A ciphersuites; if |ciphersuite_head| is nullptr, then the
- *  AES-128-GCM-SHA256 ciphersuite is selected by default. **/
 struct grpc_s2a_credentials_options {
-  const struct grpc_s2a_credentials_options_vtable* vtable;
-  s2a_ciphersuite* ciphersuite_head;
+ public:
+  ~grpc_s2a_credentials_options();
+
+  /** Setter methods. They do not take ownership of the arguments, nor do they
+   * remove a duplicate ciphersuite or target service account from the
+   * appropriate vector, if a duplicate exists. **/
+  void set_handshaker_service_url(const char* handshaker_service_url);
+  void add_supported_ciphersuite(uint16_t ciphersuite);
+  void add_target_service_account(const char* target_service_account);
```

Comment:
Maybe adding comments saying that this API should be called only at the client side?

Reply:
Done.